src/Symfony/Component/Messenger/Bridge/Amqp/Transport/AmqpStamp.php

```diff
@@ -45,7 +46,7 @@ public function getAttributes(): array
         return $this->attributes;
     }
 
-    public static function createFromAmqpEnvelope(\AMQPEnvelope $amqpEnvelope, self $previousStamp = null): self
+    public static function createFromAmqpEnvelope(\AMQPEnvelope $amqpEnvelope, self $previousStamp = null, string $retryRoutingKey = null): self
```

Comment:
We are using `$amqpReceivedStamp->getQueueName()` to populate the `$retryRoutingKey`, is that really correct? 

A routing key is not the same as the queue name. 

(It is probably correct, but I just need you to clarify.)

Reply:
This is correct. We want to distinguish:
* delay scenario, where message should be sent to original exchange to original routing key (so that all consumers can get it)
* retry scenario, where message should be sent to default (direct) exchange to routing key which matches queue name (in this case it will be received in one queue only)

Reply:
Are we sure the routing key always matches the queue name?

Reply:
For direct exchange in AMQP 0-9-1 yes:
> The default exchange is a direct exchange with no name (empty string) pre-declared by the broker. It has one special property that makes it very useful for simple applications: every queue that is created is automatically bound to it with a routing key which is the same as the queue name.

See https://www.rabbitmq.com/tutorials/amqp-concepts.html

Reply:
@Nyholm can you review please?